library/src/scripts/forms/buttonStyles.ts

```diff
@@ -276,13 +283,78 @@ export const buttonVariables = useThemeCache((forcedVars?: IThemeVariables) => {
     };
 });
 
-export const buttonSizing = (minHeight, minWidth, fontSize, paddingHorizontal, formElementVars, debug?: boolean) => {
-    const borderWidth = formElementVars.borders ? formElementVars.borders : buttonGlobalVariables().border.width;
+export interface IPaddingBasedOnRadius {
+    radius?: number | string; // pixel value in integer or percentage in string
+    extraPadding?: number | string; // If undefined, return 0 for both sides
+    height?: number; // Min height of element
+    side?: "horizontal" | "left" | "right"; // defaults to horizontal
+    debug?: boolean;
+}
+
+export const getPaddingOffsetBasedOnBorderRadius = (props: IPaddingBasedOnRadius) => {
```

Comment:
Looking at this function and it's usage outside buttons, I think we should use move this it's own file (or maybe the input styles?)